### ISSUE TITLE
/wa menu is not loading.

### ISSUE THREAD

**Issue by Gambunt** [WeakAuras/WeakAuras2#1107]
**Describe the bug**
I'm currently facing the following problem after I shared a WA I created today:
As soon as I press /wa the menu pops and the icons are not loading. It's stuck on 1/773 like the picture shown in attached files. I tried reinstaling the addon, but it didn't work. I also posted my problem on help-2 discord channel and I got an answer from rivers. After I downloaded BugShack, my error was integer overflow and I was told to upload my SavedVariables folder here.

**Do you have an error log of what happened?**
Everything is on the attached files.

**To Reproduce**
I cannot replicate it, as I cannot even open my WAs menu.

**Screenshots**
Attached

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**
I disabled everything when I tested BugShack.

**Which version of WeakAuras are you using?**
The latest one on 17/1,

**Was it working in a previous version? If yes, which was the last good one?**
It was working on this version too, up until the point that it just crushed.

**Additional context**
Attached.
[WAs.zip](https://github.com/WeakAuras/WeakAuras2/files/2766825/WAs.zip)

**Comment by emptyrivers** [comment]
The zip file you attached does not contain the files I asked for. You can find them at `World of Warcraft/_retail_/WTF/Account/<account #>/SavedVariables/WeakAuras.lua` (change <account #> to the relevant account number).

**Comment by Gambunt** [comment]
Done, my bad.
[WAs.zip](https://github.com/WeakAuras/WeakAuras2/files/2767265/WAs.zip)

**Comment by Gambunt** [comment]
I seem to have located the problem, but I cannot troubleshoot it, so I'm gonna give you the info I found:
I basically searched through the WeakAuras.lua file to see how it works and I found where each of my WAs are stored. I went then and I deleted (while keeping a backup) the WAs I played with some basic LUAs yesterday and everything else worked fine as if nothing else had happened, meaning that the problematic case was found.
The said WA I created was called "0MekkatorqueTEST", which is the parent WA containing 5 children WAs: "CrossButton", "SquareButton", "StarButton", "TriangleButton" and "DiamondButton". You can search for those in the uploaded file, but I'll be attaching those alone here in the following pastebin:
https://pastebin.com/qmtRjFqG

I'm pretty sure the problem is on the Custom Action I wrote as I was playing for the first time with LUAs, which is posted after this paragraph. The same code exists on all 5 children for 5 different macros (for chat output), whenever you click on any of the 5 different WAs. It basically shows you 5 icons and links the said chat icon after you click on any of them. It's a communication tool I wanted to use for the upcomming Mekkatorque fight for non-communicating groups to ease the whole procedure out, which I think is a great tool for any pug raider, or any raider without a working microphone. I was going to upload this into my wago.io profile after the release of the raid, which I needed to get into some SpellIDs for loading options and IconIDs. Alas, my plans were foiled after the error I encountered, which made me very frustrated and drove me mad. I'd love to hear your feedback on what the exact problem on my code is and how to fix it. Appreciate your time on this :)


This is one of the 5 buttons code. DiamondButton example here. All others are the same:

button = CreateFrame(\"Button\", \"crossButton\", WeakAuras.regions['DiamondButton'].region, \"SecureActionButtonTemplate\")\n
button:SetAllPoints()\n
button:SetAttribute(\"type1\", \"macro\")\n
button:SetAttribute(\"macrotext1\", \"/s {diamond}\")\n\n\n",

**Comment by asakawa-k** [comment]
I can't see an immediate reason why that would have caused an integer overflow so this comment is something of an aside, just to say that what you're trying to do there is a bad idea. WeakAuras isn't set up to have its Auras parent to secure frames so you're liable to get some unintended results. In this case a secure button isn't even needed since sending a chat message isn't protected. 

A couple of other minor things, you're setting `button` as global in each. and specifying the Aura ID in the region rather than using `aura_env.id` which wouldn't have issues if the Aura's name changed for any reason. 

Since all of this isn't directly related to your ticket/issue, rather than discussing it any further here, feel free to ping me on the Discord to discuss more general ways to handle making buttons for that fight.

**Comment by emptyrivers** [comment]
You correctly identified the aura causing the issue, but not the issue itself. the problem was that the CrossButton Aura that you created had an absurdly large spell id (hence the integer overflow). 
[Here](https://github.com/WeakAuras/WeakAuras2/files/2776007/WeakAuras.zip) is a repaired version of your SV file.

This is still a bug that we probably should fix, though. So I'll leave this ticket open. =

**Comment by Gambunt** [comment]
Thanks a lot. Apparently there are many different "Enlisted" buffs for when you open war mode and a party member I wanted to share my WA with, had one with a different ID. So I went into the trigger function to add that ID and shortly after I shared it with him and he got an error. I reloaded my game and the issue is known after that. I cannot recall adding any huge number on the Spell ID box other than the 6 digit Enlisted ID, but I remember changing the spell ID on the parent WA, instead of all the 5 children one by one and the problem was enountered ONLY on the CrossButton WA and not on the other 4. Maybe this specific info will help you as a form of feedback.
Of course all this is done for testing purposes, as I'm waiting for the raid's release to check specific spell IDs for the WA to show during specific parts of the encounter.
Take care :)
